main.nf

```diff
@@ -1,643 +1,286 @@
+        generate_sheets( bbi_universal_sheet_file )
+
+        sample_sheet_file = generate_sheets.out.sample_sheet
+    } else {
+        check_sample_sheet( sample_sheet_file, star_file, rt_barcode_file, params.level, params.max_wells_per_sample )
+        
+        sample_sheet_file = check_sample_sheet.out.good_sample_sheet
+    }
+
+    if ( !params.run_recovery ) {
+        make_sample_sheet( run_parameters_file, sample_sheet_file )
+
+        bcl_sample_sheet = make_sample_sheet.out.bcl_sample_sheet
+    }
+
+    if (params.max_cores > 16) {
+        max_cores_bcl = 16
+        bcl_mem = params.bcl_max_mem/16
+    } else {
+        max_cores_bcl = params.max_cores
+        bcl_mem = params.bcl_max_mem/max_cores_bcl
+    }
+
+    bcl2fastq(
+        run_dir, 
+        bcl_sample_sheet, 
+        max_cores_bcl, 
+        bcl_mem ,
+        params.bcl2fastq_barcode_mismatches,
+        params.minimum_read_length_after_trim
+    )
+
+    fastqs = bcl2fastq.out.fastqs.flatten()
 
-    input:
-        set file(R1), file(R2) from fastqs_path1
-        file sample_sheet_file1
+    /*
+    ** ================================================================================
+    ** PATH 1 - For small datasets, no fastq splitting
+    ** ================================================================================
+    */
+    if (!params.large) {
+        seg_sample_fastqs1(
+            fastqs,
+            run_parameters_file,
+            sample_sheet_file,
+            rt_barcode_file,
+            p5_barcode_file,
+            p7_barcode_file,
+            lig_barcode_file
+        )
+
+        csv_stats = seg_sample_fastqs1.out.csv_stats
+        json_stats = seg_sample_fastqs1.out.json_stats.flatten()
+    }
 
-    output:
-        file "demux_out/*" into seg_output1
-        file "demux_out/*.fastq.gz" into samp_fastqs_check1
-        file "demux_out/*.stats.json" into json_stats1 mode flatten
-        file "demux_out/*.csv" into csv_stats1
+    /*
+    ** ================================================================================
+    ** PATH 2 - For large datasets, with fastq splitting
+    ** ================================================================================
+    */
+    else {
+        fastq_chunks = fastqs.splitFastq(by: params.fastq_chunk_size, file: true, pe: true)
+        seg_sample_fastqs2( 
+            fastq_chunks,
+            run_parameters_file,
+            sample_sheet_file,
+            rt_barcode_file,
+            p5_barcode_file,
+            p7_barcode_file,
+            lig_barcode_file
+        )
+
+        all_csv_stats = seg_sample_fastqs2.out.csv_stats.flatten()
+        all_json_stats = seg_sample_fastqs2.out.json_stats.flatten()
+        sample_fastqs_check = seg_sample_fastqs2.out.sample_fastqs_check.flatten()
+
+        // Combine Fastqs
+        get_prefix = { fname ->
+            (fname - ~/_R1_001\.[0-9]+\.fastq.fastq.gz/)
```

Comment:
I'm not sure if you can make this level of modification to the underlying workflow, but this line makes me think that there may be some amount of using file names to keep track of which files are from which specimens. As a general principal, filenames should never be used to encode things like specimen, etc., since that has one of the largest possibilities for introducing perplexing errors in the future. That's just a more general note on workflow design, no specific action needed necessarily.